core/src/gl/texture.cpp

```diff
@@ -155,18 +155,16 @@ void Texture::update(GLuint _textureUnit) {
 
         m_shouldResize = false;
 
-        m_subData.clear();
+    }
 
```

Comment:
Actually we dont need the subdata at all since the updated area is copied to to m_data in setSubData. It only needs to keep track of the  y-start and y-end and copy that region to the texture in Texture::update. This will also avoid copying overlapping subdata regions multiple time to the gl texture